transit_odp/publish/forms.py

```diff
@@ -554,3 +607,122 @@ def get_layout(self):
+
+    def get_layout(self):
+        return Layout(
+            Div("licence", css_class="licence"),
+            SEPARATOR,
+            ButtonHolder(
+                CONTINUE_BUTTON,
+                cancel_seasonal_service(self.org_id),
+                css_class="buttons",
+            ),
+        )
+
+
+class SeasonalServiceEditDateForm(GOVUKModelForm):
+    form_tag = False
+    form_error_title = DEFAULT_ERROR_SUMMARY
+
+    registration_code = forms.IntegerField(
+        required=True,
+        label=_("Service code"),
+        error_messages={"required": _("Enter a service code in the box below")},
+    )
+    start = forms.DateTimeField(
+        required=True,
+        label=_("Service begins on"),
+        error_messages={"invalid": _("Error first date")},
+        widget=NumberInput(attrs={"type": "date"}),
+    )
+    end = forms.DateTimeField(
+        required=True,
+        label=_("Service ends on"),
+        error_messages={"invalid": _("Error last date")},
+        widget=NumberInput(attrs={"type": "date"}),
+    )
+
+    class Meta:
+        model = SeasonalService
+        fields = ("registration_code", "start", "end")
+
+    def __init__(self, *args, **kwargs):
+        self.licence = kwargs.pop("licence", None)
+        self.org_id = kwargs.pop("org_id")
+        super().__init__(*args, **kwargs)
+
+    def get_layout(self):
+        help_modal = render_to_string(
+            "publish/snippets/help_modals/seasonal_services.html"
+        )
+        return Layout(
+            Field(
+                "registration_code",
+                template="publish/seasonal_services/service_code_widget.html",
+            ),
+            HTML(
+                format_html(
+                    '<h2 class="govuk-heading-s">Service operating dates {}</h2>',
+                    help_modal,
+                )
+            ),
+            DateDiv(
+                FieldNoErrors("start", wrapper_class="date"),
+                FieldNoErrors("end", wrapper_class="date"),
+                css_class="date-container",
+            ),
+            SEPARATOR,
+            ButtonHolder(
+                CONTINUE_BUTTON,
+                cancel_seasonal_service(self.org_id),
+                css_class="buttons",
+            ),
+        )
+
+    def clean_registration_code(self):
+        registration_code = self.cleaned_data["registration_code"]
+        exists = SeasonalService.objects.filter(
+            licence=self.licence, registration_code=registration_code
+        ).exists()
+        if exists:
+            raise ValidationError(
+                "This service code has already been set up with a date range"
+            )
+        return registration_code
+
+    def clean(self):
+        cleaned_data = self.cleaned_data
+        start = cleaned_data.get("start")
+        end = cleaned_data.get("end")
+        if start is None or end is None:
+            return cleaned_data
+
+        if start >= end:
```

Comment:
It's probably valid for them to be equaL, no? i.e. a season that lasts just one day.

Reply:
Good question! I dont know sounds plausible when you say it like this. Ill change.